src/test/java/com/wb/WeatherForecastApiApplicationUnitTests.java

```diff
@@ -1,85 +1,54 @@
-import org.junit.runner.RunWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
+import static org.springframework.test.web.servlet.ResultMatcher.matchAll;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
 
 import com.wb.payload.WeatherMapper;
 import com.wb.service.WeatherService;
 
-@RunWith(SpringRunner.class)
 @SpringBootTest
+@AutoConfigureMockMvc
 public class WeatherForecastApiApplicationUnitTests {
-    @Mock
-    private WeatherService weatherService;
-    @Mock
-    Future<WeatherMapper> futureWeather = new CompletableFuture<>();
-    @InjectMocks
-    WeatherForecastController controller = new WeatherForecastController(weatherService);
-    WeatherMapper weatherMapper;
 
-    @org.junit.jupiter.api.Test
-    public void contextLoads() {
-    }
+    private final WeatherService weatherService;
+    private final MockMvc mockMvc;
 
-    @Before
-    public void setUp1() throws Exception {
-        weatherMapper = new WeatherMapper();
-        when(futureWeather.isDone()).thenReturn(true);
-        when(futureWeather.get()).thenReturn(weatherMapper);
-        when(weatherService.fetchWeatherByCity(Mockito.anyString())).thenReturn(weatherMapper);
-        when(weatherService.testWeatherByCity(Mockito.anyString())).thenReturn(futureWeather);
+    @Autowired
+    public WeatherForecastApiApplicationUnitTests(WeatherService weatherService, MockMvc mockMvc) {
+        this.weatherService = weatherService;
+        this.mockMvc = mockMvc;
     }
 
     @Test
-    public void shouldReturnCityWeatherObject() throws InterruptedException, ExecutionException {
-        Future<WeatherMapper> fw = weatherService.testWeatherByCity("Warsaw");
-        while (!fw.isDone()) {
-            Thread.sleep(10);
-        }
-        assertEquals(weatherMapper, fw.get());
-    }
-
-    @Before
-    public void setUp2() throws Exception {
-        weatherMapper = new WeatherMapper();
-        when(futureWeather.isDone()).thenReturn(true);
-        when(futureWeather.get()).thenReturn(weatherMapper);
-        when(weatherService.fetchWeatherByCityAndTime(Mockito.anyString(), Mockito.anyLong(), Mockito.anyLong())).thenReturn(weatherMapper);
-        when(weatherService.testWeatherByCityAndTime(Mockito.anyString(), Mockito.anyLong(), Mockito.anyLong())).thenReturn(futureWeather);
+    public void apiFetchWeatherByCity() throws Exception {
```

Comment:
jeśli wyrzucisz te test i `apiFetchWeatherByCityAndTime` (bo nic nie robią te testy) to klasa stanie sie ala testem integracyjny klasy `WeatherService` zupelnie zbednym bo test integracyjny aplikacji calej juz masz w innej klasie